lib/rubocop/cop/performance/zip_for_array_wrapping.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # Checks for `.map { |id| [id] }` and suggests replacing it with `.zip`.
+      #
+      # @example
+      #   # bad
+      #   [1, 2, 3].map { |id| [id] }
+      #
+      #   # bad
+      #   [1, 2, 3].map { [_1] }
+      #
+      #   # good
+      #   [1, 2, 3].zip
+      #
+      # @example
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| id }
+      #
+      # @example
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| [id, id] }
+      class ZipForArrayWrapping < Base
+        include RangeHelp
+        extend AutoCorrector
+
+        MSG = 'Use `zip` instead of `%<original_code>s`.'
+        RESTRICT_ON_SEND = Set.new(%i[map collect]).freeze
+
+        # Matches regular block form `.map { |e| [e] }`
+        # @!method map_with_array?(node)
+        def_node_matcher :map_with_array?, <<~PATTERN
+          (block
+            (send _ RESTRICT_ON_SEND)
+            (args (arg _id))
+            (array (lvar _id)))
+        PATTERN
+
+        # Matches numblock form `.map { [_1] }`
+        # @!method map_with_array_numblock?(node)
+        def_node_matcher :map_with_array_numblock?, <<~PATTERN
+          (numblock
+            (send _ :map)
+            1
+            (array (lvar _))
+          )
+        PATTERN
```

Comment:
You can combine these matchers into a single one:

```suggestion
        # @!method map_with_array?(node)
        def_node_matcher :map_with_array?, <<~PATTERN
          {
            (block (send _ RESTRICT_ON_SEND) (args (arg _id)) (array (lvar _id)))
            (numblock (send _ RESTRICT_ON_SEND) 1 (array (lvar _))
            )
          }
        PATTERN
```

Note that also you previously were hardcoding `:map` for `numblock`s which meant `[1, 2, 3].collect { [_1] }` wasn't being detected. I replaced it with `RESTRICT_ON_SEND` above. Please add a test for this case too.